bve-corpus/src/logger.rs

```diff
@@ -25,36 +33,48 @@ struct Panic {
     cause: String,
 }
 
-pub fn receive_results(options: &Options, result_source: &Receiver<FileResult>) {
+pub fn receive_results(options: &Options, result_source: Receiver<FileResult>) {
     let mut results = ResultCollection::default();
 
     while let Ok(result) = result_source.recv() {
+        let single_file_result = results.file_types.entry(result.kind).or_default();
         match result.result {
-            ParseResult::Success => results.successes.push(result.path),
-            ParseResult::Errors { count, error } => results.failures.push(Failure {
+            ParseResult::Success => single_file_result.successes.push(result.path),
+            ParseResult::Errors { count, error } => single_file_result.failures.push(Failure {
                 error: error.to_string(),
                 count,
                 path: result.path,
             }),
-            ParseResult::Panic { cause } => results.panics.push(Panic {
+            ParseResult::Panic { cause } => single_file_result.panics.push(Panic {
                 cause,
                 path: result.path,
             }),
-            ParseResult::Finish => break,
+            ParseResult::Finish => {
+                std::mem::forget(result_source); // We're finishing, we don't actually care about if this is cleaned up, and this prevents a out-of-time panic cascade 
+                break;
+            }
         }
     }
 
-    results.failures.retain(|v| v.count != 0);
+    let (panics, failures, successes) = results
+        .file_types
+        .values_mut()
+        .map(|single| {
+            single
+                .failures
+                .sort_by_cached_key(|v| Reverse((v.count, v.path.clone())));
 
-    results
-        .failures
-        .sort_by_cached_key(|v| Reverse((v.count, v.path.clone())));
+            (single.panics.len(), single.failures.len(), single.successes.len())
+        })
+        .fold((0, 0, 0), |(acc_p, acc_f, acc_s), (p, f, s)| {
```

Comment:
What are these variable names?

Reply:
I'm trying to implement summing a 3 tuple because they don't implement `Add` for good reason. `acc` is just the accumulator and p/f/s is panics/failures/successes. I know it's not particularly readable, but I was just trying to get things working and didn't feel like writing long ass names for a closure that needs 6 input variables.